metric_namer.go

```diff
@@ -96,10 +121,24 @@ type Metric struct {
 
 // Build builds a metric name for the specified metric.
 //
-// If UTF8Allowed is true, the metric name is returned as is, only with the addition of type/unit suffixes and namespace preffix if required.
-// Otherwise the metric name is normalized to be Prometheus-compliant.
-// See rules at https://prometheus.io/docs/concepts/data_model/#metric-names-and-labels,
-// https://prometheus.io/docs/practices/naming/#metric-and-label-naming
+// The method applies different transformations based on the MetricNamer configuration:
+//   - If UTF8Allowed is true, preserves UTF-8 characters
+//   - If UTF8Allowed is false, replaces invalid characters with underscores
```

Comment:
I think the original wording is imprecise. E.g. it doesn't define invalid characters.
```suggestion
If UTF8Allowed is false, translates metric names to comply with legacy Prometheus name scheme
```